tutorials/yes-no-game/full game/yesnogame.edn

```diff
@@ -0,0 +1,251 @@
+(def max-timer 5)
+
+(defshards initialize []
+  (LoadImage "data/cats/cat01.png") (Push :Name .cat-images)
+  (LoadImage "data/cats/cat02.png") (Push :Name .cat-images)
+  (LoadImage "data/cats/cat03.png") (Push :Name .cat-images)
+  (LoadImage "data/cats/cat04.png") (Push :Name .cat-images)
+  (LoadImage "data/cats/cat05.png") (Push :Name .cat-images)
+  (LoadImage "data/dogs/dog01.png") (Push :Name .cat-images)
+  (LoadImage "data/dogs/dog02.png") (Push :Name .dog-images)
+  (LoadImage "data/dogs/dog03.png") (Push :Name .dog-images)
+  (LoadImage "data/dogs/dog04.png") (Push :Name .dog-images)
+  (LoadImage "data/dogs/dog05.png") (Push :Name .dog-images)
+  (LoadImage "data/penguins/penguin01.png") (Push :Name .penguin-images)
+  (LoadImage "data/penguins/penguin02.png") (Push :Name .penguin-images)
+  (LoadImage "data/penguins/penguin03.png") (Push :Name .penguin-images)
+  (LoadImage "data/penguins/penguin04.png") (Push :Name .penguin-images)
+  (LoadImage "data/penguins/penguin05.png") (Push :Name .penguin-images)
+  .cat-images (Push  :Name .all-images)
+  .dog-images (Push :Name .all-images)
+  .penguin-images  (Push :Name .all-images)
+
+  0 >= .left-series-index
+  0 >= .right-series-index
+  0 >= .left-image-index
+  0 >= .right-image-index
+
+  (Count .all-images) >= .total-image-series
+  0 >= .left-images-count
+  0 >= .right-images-count
+
+  .cat-images >= .left-images
+  .cat-images >= .right-images
+  true >= .same-images
+  true >= .new-round
+  0 >= .new-round-number
+
+  ; variables to reset each round
+  1 >= .current-round
+  0 >= .seconds-elapsed
+  0 >= .total-score
+  false >= .yes-input
+  false >= .input-received
+  false >= .game-over)
+
+(defshards reset-values []
+
+  false > .yes-input
+  false > .input-received
+  false > .game-over
+  1 > .current-round
+  0 >  .seconds-elapsed
+  0 > .total-score)
+
+; determines the images used for the round
+(defshards initialize-round []
+
+  0 > .seconds-elapsed
+  
+  ; determine if the images on the left and right will be the same or different
+  (If
+   :Predicate (-> (RandomInt :Max 2) (Is 0))
+   :Then (-> true > .same-images)
+   :Else (-> false > .same-images))
+
+  ; gets the image series to be used for the left side
+  (RandomInt :Max .total-image-series) > .left-series-index
+
+  ; get the number of images in the series to be used for the left
+  .all-images (Take .left-series-index) > .left-images
+  (Count .left-images) > .left-images-count
+
+  ; setup for the right image
+  (If
+   :Predicate (-> .same-images (Is true))
+   ; if images should be from the same series
+   :Then (->
+          .left-series-index > .right-series-index
+          (RandomInt :Max .left-images-count) > .left-image-index
+          (RandomInt :Max .left-images-count) > .right-image-index
+          (Repeat
+           :Action (-> (RandomInt :Max .left-images-count) > .right-image-index)
+           :Forever true
+           :Until (-> .right-image-index (IsNot .left-image-index)))
```

Comment:
Although rare, there is no guarantee that looping in an attempt to get a different value randomly would ever complete. It could in theory take forever.

Instead of looping, we should consider using a shuffle algorithm to randomly reorganize the two sequences once and then iterate them. I would suggest the [Fisher-Yates shuffle](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle). I have a working implementation somewhere but I'll let you do this as an exercise and then we can trade implementations.